src/components/app/routes/loaders/utils.ts

```diff
@@ -95,7 +104,207 @@ const populateCompletedFormFields = ({
+  isValidPayload: boolean,
+} => {
+  const checkoutFormData = checkoutFormStore.getState().formData;
+
+  const {
+    quantity,
+    adminEmail,
+    stripePriceId,
+  } = checkoutFormData[DataStoreKey.PlanDetails];
+  const {
+    enterpriseSlug,
+    companyName,
+  } = checkoutFormData[DataStoreKey.AccountDetails];
+
+  const checkoutSessionPayload = {
+    quantity,
+    adminEmail,
+    stripePriceId,
+    enterpriseSlug,
+    companyName,
+  };
+
+  const isPresent = v => v != null && v !== ''; // != null covers null and undefined
+
+  const isValidPayload = Object.values(checkoutSessionPayload).every(isPresent);
+
+  return {
+    checkoutSessionPayload,
+    isValidPayload,
+  };
+};
+
+/**
+ * A union of all route path values defined on CheckoutPageRoute (e.g., "/plan-details").
+ */
+type CheckoutPageRouteValue = (typeof CheckoutPageRoute)[keyof typeof CheckoutPageRoute];
+
+/**
+ * Result of validating whether navigation to a given route is allowed based on form state.
+ */
+type ValidationResult = {
+  /** Whether all prerequisites passed. */
+  valid: boolean;
+  /** If invalid, the first route the user should be redirected to fix validation. */
+  invalidRoute?: CheckoutPageRouteValue; // route of the first page that failed validation
+};
+
+/**
+ * Build React Hook Form resolvers (zod-based) for the various checkout form slices,
+ * using the provided field constraints and Stripe price ID where required.
+ *
+ * Note: Billing Details resolver can be added later when the schema is available.
+ *
+ * @param {CheckoutContextFieldConstraints} constraints - Field-level constraints from the BFF context.
+ * @param {CheckoutContextPrice['id']} stripePriceId - Stripe price identifier used by the Plan Details schema.
+ * @returns {{ planDetailsResolver: Function, accountDetailsResolver: Function }} An object with resolvers.
+ */
+const makeResolvers = (
+  constraints: CheckoutContextFieldConstraints,
+  stripePriceId: CheckoutContextPrice['id'],
+) => {
+  const planDetailsResolver = zodResolver(PlanDetailsSchema(constraints, stripePriceId));
+
+  const accountDetailsResolver = zodResolver(AccountDetailsSchema(constraints));
+
+  // const billingDetailsResolver: ResolverFn =
+  //   zodResolver(BillingDetailsSchema(constraints));
+
+  return {
+    planDetailsResolver,
+    accountDetailsResolver,
+    // billingDetailsResolver,
+  };
+};
+
+interface PrerequisiteCheck<T> {
+  pick: (formData: any) => T;
+  getResolver: (
+    constraints: CheckoutContextFieldConstraints, stripePriceId: CheckoutContextPrice['id']
+  ) => (values: any, ctx?: any, opts?: any) => any;
+  failRoute: CheckoutPageRouteValue;
+}
+
+/**
+ * For each target route, list the prerequisite slices to validate, in order.
+ * Each entry includes which form slice to validate, how to build its resolver,
+ * and the route that should be returned if that slice is invalid.
+ */
+const prerequisiteSpec: Record<string, Array<PrerequisiteCheck<any>>> = {
```

Comment:
be more explicit with the left-hand-side of the Record:
```suggestion
const prerequisiteSpec: Record<CheckoutStep, Array<PrerequisiteCheck<any>>> = {
```